components/inputs/input-date.js

```diff
@@ -469,6 +469,7 @@ class InputDate extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixin(
 			this._openedOnKeydown = true;
 			this.opened = true;
 			e.preventDefault();
+			e.stopPropagation();
```

Comment:
Seems reasonable. It has me contemplating whether there is a more appropriate place to handle this. Some place in filter, the list, or maybe `SelectionMixin` is better. Thoughts?

Reply:
Yup that's possible - I'll test some other options out

Reply:
Based on some testing, I'm not sure this is possible (unless I'm missing something here) - I looked at doing this in both `FilterDimensionSetDateTimeRangeValue` and `list-item-generic-layout` and looked at what the target of the `keydown` event was. The problem was that the target would be the `d2l-input-date-range` (or date-time range) component if the down arrow happened when focus on `input-date` (where we'd want to prevent down arrow from going to the next list item) or if was on the `InteractiveMixin` wrapping the range input (where we wouldn't want to prevent the down arrow from going to the next list item).

We could put this in `InteractiveMixin` [here](https://github.com/BrightspaceUI/core/blob/main/mixins/interactive/interactive-mixin.js#L127) and handle it like we do tab though I also don't know if that's a great place for it since it feels like it's not really `InteractiveMixin`'s problem.

Reply:
What's causing the focus to move? Is it the focus behaviour provided by [d2l-list-item-generic-layout](https://github.com/BrightspaceUI/core/blob/main/components/list/list-item-generic-layout.js)? Or is it the radio logic provided by [SelectionMixin](https://github.com/BrightspaceUI/core/blob/main/components/selection/selection-mixin.js#L156). I'm guessing it's the former, since the `SelectionMixin` appears to check the event target before proceeding to move the focus to the next/previous radio.

For the `InteractiveMixin` containing tags, how do we handle the arrow keys? Do we stop the propagation for those?

Reply:
The focus behaviour is provided by the list item generic layout (since list is using `selectionNoInputArrowKeyBehaviour` so it's early exiting from the SelectionMixin focus behaviour). We could do something similar to what SelectionMixin is doing, maybe check if the target is within `InteractiveMixin` then ignore (though that maybe gets a little too much into what's contained in the list item?). It's a little more complex than `SelectionMixin`, which can just check for that class.

For tags, they don't look to be doing anything special within `InteractiveMixin`. They use `ArrowKeysMixin` which does a `e.stopPropagation` and that seems to be enough.

Reply:
Ah ok, that makes sense for tags.

Sounds like handling in `SelectionMixin` isn't going to help us since it's the `d2l-list-item-generic-layout` that's incorrectly moving the focus.

Handling in `d2l-input-date` like you have here is akin to relying on `ArrowKeysMixin`, in the sense that it's the component contained within the "interactive container" that's stopping the propagation. 

Are there other components we might need to worry about? More generally, if a consumer puts a `d2l-input-text` inside of an interactive container and the user presses the arrow keys while the focus is within, does that cause a problem? If so, perhaps stopping the propagation more generally within `InteractiveMixin` provides a more comprehensive fix.

Reply:
That's a good point. In theory, it's possible especially if the consumer would want arrow keys to do something else in `InteractiveMixin`. If an intention of `InteractiveMixin` was to deal with things like this then it does sound like it would make sense to handle this there (as well as other keys, e.g., up arrow, I'm assuming)

Reply:
Yeah, the first usage was `d2l-tag-list`. And while we envisioned that people might eventually want to put other things inside, we didn't develop those cases, so I'm not totally surprised that we need to enhance it to cover more scenarios.